Assets/HoloToolkit/Build/Editor/BuildDeployWindow.cs

```diff
@@ -0,0 +1,581 @@
+        private string devicePassword = "";
+        private bool fullReinstall = true;
+
+        // Functions
+        [MenuItem("HoloToolkit/Build Window", false, 101)]
+        public static void OpenWindow()
+        {
+            BuildDeployWindow window = GetWindow<BuildDeployWindow>("Build Window") as BuildDeployWindow;
+            if (window != null)
+            {
+                window.Show();
+            }
+        }
+
+        void OnEnable()
+        {
+            Setup();
+        }
+
+        private void Setup()
+        {
+            this.titleContent = new GUIContent("Build Window");
+
+            // Query menu defaults
+            buildDirectory = GetEditorPref(EditorPrefs_BuildDir, buildDirectory);
+            msBuildVer = GetEditorPref(EditorPrefs_MSBuildVer, msBuildVer);
+            buildConfig = GetEditorPref(EditorPrefs_BuildConfig, buildConfig);
+            forceRebuildAppx = (GetEditorPref(EditorPrefs_ForceRebuild, forceRebuildAppx ? "true" : "false") == "true");
+            targetIPs = GetEditorPref(EditorPrefs_CustomIP, targetIPs);
+            deviceUser = GetEditorPref(EditorPrefs_DeviceUser, deviceUser);
+            devicePassword = GetEditorPref(EditorPrefs_DevicePwd, devicePassword);
+            fullReinstall = GetEditorPref(EditorPrefs_FullReinstall, fullReinstall);
+
+            this.minSize = new Vector2(600, 200);
+
+            UpdateBuilds();
+        }
+
+        private void OnGUI()
+        {
+            GUILayout.Space(GUISectionOffset);
+
+            // Setup
+            int buttonWidth_Quarter = Screen.width / 4;
+            int buttonWidth_Half = Screen.width / 2;
+            int buttonWidth_Full = Screen.width - 25;
+            string appName = PlayerSettings.productName;
+
+            // Build section
+            GUILayout.BeginVertical();
+                GUILayout.Label("SLN");
+
+                EditorGUILayout.BeginHorizontal();
+                GUILayout.FlexibleSpace();
+                EditorGUILayout.EndHorizontal();
+
+                // Build directory (and save setting, if it's changed)
+                string newBuildDirectory = EditorGUILayout.TextField(GUIHorizSpacer + "Build directory", buildDirectory);
+                if (newBuildDirectory != buildDirectory)
+                {
+                    EditorPrefs.SetString(EditorPrefs_CustomIP, newBuildDirectory);
+                    buildDirectory = newBuildDirectory;
+                }
+
+                // Build SLN button
+                using (new EditorGUILayout.HorizontalScope())
+                {
+                    GUILayout.FlexibleSpace();
+                    GUI.enabled = ShouldOpenSLNBeEnabled;
+                    if (GUILayout.Button("Open SLN", GUILayout.Width(buttonWidth_Quarter)))
+                    {
+                        // Open SLN
+                        FileInfo slnFile = new FileInfo(Path.Combine(buildDirectory, PlayerSettings.productName + ".sln"));
```

Comment:
Acts funky if no solution exists. Maybe disable in that case? Current Build menu asks if you'd like to build one in this case.